pypesto/ensemble/ensemble.py

```diff
@@ -367,6 +371,155 @@ def from_sample(
         x_vectors = x_vectors.T
         return Ensemble(x_vectors, **kwargs)
 
+    @staticmethod
+    def from_optimization_endpoints(
+            result: Result,
+            cutoff: float = np.inf,
+            max_size: int = np.inf,
+            **kwargs,
+    ):
+        """Construct an ensemble from an optimization result.
+
+        Parameters
+        ----------
+        result:
+            A pyPESTO result that contains an optimization result.
+        cutoff:
+            Exclude parameters from the optimization if the
+            nllh is higher than the `cutoff`.
+        max_size:
+            The maximum size the ensemble should be.
+
+        Returns
+        -------
+        The ensemble.
+        """
+        x_vectors = []
+        vector_tags = []
+
+        for start in result.optimize_result.list:
```

Comment:
yes, as the minimize function has an inherent sorting here (https://github.com/ICB-DCM/pyPESTO/blob/625c41911fdee2608e736321d9e9050ac90f7ba4/pypesto/optimize/optimize.py#L119) I assumed that the optimization results should all be sorted.

Reply:
I am not sure whether we always make sure that the values are sorted, e.g. when loading from file / appending. it may be useful to, as I think we do now in visualization, always explicitly sort when the n best points or the order is needed.